plugins/net_plugin/net_plugin.cpp

```diff
@@ -3847,7 +3841,7 @@ namespace eosio {
 
       uint32_t lib = cc.last_irreversible_block_num();
       try {
-         if( blk_num <= lib || cc.block_exists(blk_id) ) {
+         if( blk_num <= lib || cc.validated_block_exists(blk_id) ) {
```

Comment:
why `validated`?

Reply:
Because `cc.block_exists(blk_id)` is always true here. We added to the fork_db when the block_state was created. However, if validated then that means its state has been applied we don't need this block.